File: SurTerrain/src/Models/Joueurs.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package Models;

import Services.Serviceclub;

/**
 *
 * @author aissa
 */
public class Joueurs {

    private int id, age, id_club;
    private String nom, email;
    private String prenom, nom_club;

    public Joueurs(String nom, String prenom, int age, int id_club, String email) {

        this.nom = nom;
        this.prenom = prenom;
        this.age = age;
        this.id_club = id_club;
        this.email = email;

    }

    public Joueurs(int id, String nom, String prenom, int age, int id_club, String email) {
        this.id = id;
        this.nom = nom;
        this.prenom = prenom;
        this.age = age;
        this.id_club = id_club;
        Serviceclub x = new Serviceclub();
        this.nom_club = x.getnomclub(this.id_club);
        this.email = email;
    }

    public Joueurs(int age, String nom, String prenom, String email) {
        this.age = age;
        this.nom = nom;
        this.email = email;
        this.prenom = prenom;
    }
    

    public Joueurs() {
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getNom() {
        return nom;
    }

    public void setNom(String nom) {
        this.nom = nom;
    }

    public String getPrenom() {
        return prenom;
    }

    public void setPrenom(String prenom) {
        this.prenom = prenom;
    }

    public int getAge() {
        return age;
    }

    public void setAge(int age) {
        this.age = age;
    }

    public int getId_club() {
        return id_club;
    }

    public void setId_club(int id_club) {
        this.id_club = id_club;
    }

    public String getNom_club() {
        return nom_club;
    }

    public void setNom_club(String nom_club) {
        this.nom_club = nom_club;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    @Override
    public String toString() {
        return "Joueurs{" + "id=" + id + ", age=" + age + ", id_club=" + id_club + ", nom=" + nom + ", prenom=" + prenom + ", nom_club=" + nom_club + ", email=" + email + '}';
    }
    
    @Override
    public boolean equals(Object obj) {
        if (obj instanceof Joueurs) {
            Joueurs j = (Joueurs) obj;
            if (j.age == this.age) {

                return true;
            }
        }
        return false;
    }

}

File: src/Entity/Joueurs.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;

/**
 * Joueurs
 *
 * @ORM\Table(name="joueurs", indexes={@ORM\Index(name="fk_clubs", columns={"id_club"})})
 * @ORM\Entity
 * @ORM\Entity(repositoryClass="App\Repository\JoueursRepository")
 */
class Joueurs
{
    /**
     * @var int
     *
     * @ORM\Column(name="id", type="integer", nullable=false)
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="IDENTITY")
     */
    private $id;

    /**
     * @var string
     *
     * @ORM\Column(name="nom", type="string", length=20, nullable=false)
     * @Assert\NotBlank(
     * message = "remplissez le champ SVP"
     * )
     *  @Assert\Length(
     *      min = 3,
     *      max = 20,
     *      minMessage = "Le nom doit comporter au moins {{ limit }} caractères",
     *      maxMessage = "Le nom ne doit pas dépasser {{ limit }} caractères"
     *
     * )
     */
    private $nom;

    /**
     * @var string
     *
     * @ORM\Column(name="prenom", type="string", length=20, nullable=false)
     * @Assert\NotBlank(
     * message = "remplissez le champ SVP"
     * )
     * @Assert\Length(
     *      min = 5,
     *      max = 20,
     *      minMessage = "Le nom doit comporter au moins {{ limit }} caractères",
     *      maxMessage = "Le nom ne doit pas dépasser {{ limit }} caractères"
     *
     * )
     */
    private $prenom;

    /**
     * @var int
     *
     * @ORM\Column(name="age", type="integer", nullable=false)
     * @Assert\NotBlank(
     * message = "remplissez le champ SVP"
     * )
     * @Assert\LessThan (
     *     value = 20,
     *     message = "L'age ne doit pas dépasser 20 ans"
     * )
     */
    private $age;

    /**
     * @var string
     *
     * @ORM\Column(name="nom_club", type="string", length=11, nullable=true)
     */
    private $nomClub;

    /**
     * @var string
     *
     * @ORM\Column(name="email", type="string", length=30, nullable=false)
     * @Assert\Email(
     *       message = "The email '{{ value }}' is not a valid email."
     * )
     * @Assert\NotBlank
     * (
     * message = "remplissez le champ SVP"
     * )
     *
     */
    private $email;

    /**
     * @var \Club
     *
     * @ORM\ManyToOne(targetEntity="Club")
     * @ORM\JoinColumns({
     *   @ORM\JoinColumn(name="id_club", referencedColumnName="id_club")
     * })
     */
    private $idClub;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getNom(): ?string
    {
        return $this->nom;
    }

    public function setNom(string $nom): self
    {
        $this->nom = $nom;

        return $this;
    }

    public function getPrenom(): ?string
    {
        return $this->prenom;
    }

    public function setPrenom(string $prenom): self
    {
        $this->prenom = $prenom;

        return $this;
    }

    public function getAge(): ?int
    {
        return $this->age;
    }

    public function setAge(int $age): self
    {
        $this->age = $age;

        return $this;
    }

    public function getNomClub(): ?string
    {
        return $this->nomClub;
    }

    public function setNomClub(?string $nomClub): self
    {
        $this->nomClub = $nomClub;

        return $this;
    }

    public function getEmail(): ?string
    {
        return $this->email;
    }

    public function setEmail(string $email): self
    {
        $this->email = $email;

        return $this;
    }

    public function getIdClub(): ?Club
    {
        return $this->idClub;
    }

    public function setIdClub(?Club $idClub): self
    {
        $this->idClub = $idClub;

        return $this;
    }



}
